vscode-extension/src/providers/register.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import path = require('path');
+import * as fs from 'fs/promises';
+import * as semver from 'semver';
+import {workspace} from 'vscode';
+import {RelayExtensionContext} from '../context';
+import {RelayTextDocumentContentProvider} from './textDocumentContentProvider';
+
+export async function registerProviders(context: RelayExtensionContext) {
+  const relayCompilerDirectory = path.join(
+    context.relayBinaryExecutionOptions.binaryPath,
+    '../..',
+  );
+
+  let packageManifest: any;
+
+  try {
+    packageManifest = JSON.parse(
+      await fs.readFile(
+        path.join(relayCompilerDirectory, 'package.json'),
+        'utf-8',
+      ),
+    );
+  } catch {
+    // If we can't read this file, we don't have to bother continuing.
+    return;
+  }
+
+  const isSemverRangeSatisfied = semver.satisfies(
+    packageManifest.version,
+    '>=14.2',
+  );
+
+  const isPrerelease = semver.prerelease(packageManifest.version) != null;
+
+  if (!isSemverRangeSatisfied && !isPrerelease) {
```

Comment:
I'm not sure I'm following... We have to read the packageJson file above to find the version to do the semver checks. If the version doesn't match we don't register the provider and it will never attempt to read the JSON schema.